pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/channel/SetChannelMetadata.ios.kt

```diff
@@ -1,9 +1,59 @@
 package com.pubnub.api.endpoints.objects.channel
 
+import cocoapods.PubNubSwift.PubNubObjC
+import cocoapods.PubNubSwift.setChannelMetadataWithChannel
 import com.pubnub.api.Endpoint
+import com.pubnub.api.models.consumer.objects.channel.PNChannelMetadata
 import com.pubnub.api.models.consumer.objects.channel.PNChannelMetadataResult
+import com.pubnub.api.onFailureHandler
+import com.pubnub.api.onSuccessHandler
+import com.pubnub.api.v2.callbacks.Consumer
+import com.pubnub.api.v2.callbacks.Result
+import com.pubnub.kmp.CustomObject
+import kotlinx.cinterop.ExperimentalForeignApi
+import platform.posix.stat
 
 /**
  * @see [PubNub.setChannelMetadata]
  */
-actual interface SetChannelMetadata : Endpoint<PNChannelMetadataResult>
\ No newline at end of file
+actual interface SetChannelMetadata : Endpoint<PNChannelMetadataResult>
+
+@OptIn(ExperimentalForeignApi::class)
+class SetChannelMetadataImpl(
+    private val pubnub: PubNubObjC,
+    private val channel: String,
+    private val name: String?,
+    private val description: String?,
+    private val custom: CustomObject?,
+    private val includeCustom: Boolean,
+    private val type: String?,
+    private val status: String?
+): SetChannelMetadata {
+    override fun async(callback: Consumer<Result<PNChannelMetadataResult>>) {
+        pubnub.setChannelMetadataWithChannel(
+            channel = channel,
+            name = name,
+            description = description,
+            custom = null, // TODO: How to retrieve value from CustomObject?
```

Comment:
What should our approach be in this case? I'd like to pass a value from `CustomObject` but it's private for now

Reply:
make CustomObject.value public in iOS for now and pass it here, handling customobjects might change in the future but for now it'll do

Reply:
Done